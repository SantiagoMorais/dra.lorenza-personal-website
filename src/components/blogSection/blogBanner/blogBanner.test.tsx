import { render, screen } from "@testing-library/react"
import { BlogBanner } from "."
import blogImage from "@assets/imgs/blogSection/blogSectionImage.jpg"
import blogImagePNG from "@assets/imgs/blogSection/blogSectionImageWithoutBG.png"

describe("<BlogBanner />", () => {
    it("should render the component correctly", () => {
        render(<BlogBanner />);
        const title = screen.getByText(/Blog de Nutrologia: Saúde e Ciência/i);
        expect(title).toBeInTheDocument();
    });

    it("should render the title and subtitle correctly", () => {
        render(<BlogBanner />);
        const info = document.querySelector(".info");
        expect(info).toMatchSnapshot();
    });

    it("should render the banner image correctly", () => {
        render(<BlogBanner />);
        
        const bannerImage = document.querySelector(".banner");

        expect(bannerImage).toBeInTheDocument();
        expect(bannerImage).toHaveAttribute("src", blogImage)
    });

    it("should render the banner png image correctly", () => {
        render(<BlogBanner />);

        const bannerImage = document.querySelector(".bannerPNG");

        expect(bannerImage).toBeInTheDocument();
        expect(bannerImage).toHaveAttribute("src", blogImagePNG)
    });
})