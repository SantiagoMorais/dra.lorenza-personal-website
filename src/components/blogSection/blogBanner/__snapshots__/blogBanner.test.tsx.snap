// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<BlogBanner /> > should render the title and subtitle correctly 1`] = `
<div
  class="info"
>
  <h2
    class="title"
  >
    Blog de Nutrologia: Saúde e Ciência
  </h2>
  <h3
    class="subtitle"
  >
    Informações precisas para otimizar sua alimentação e seu bem-estar
  </h3>
</div>
`;
